Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPIBookmarks.mm

```diff
@@ -413,6 +413,64 @@ void configureUpdateBookmarksDelegate(TestWebExtensionManager *manager)
             completionHandler(updatedMockNode, nil);
         };
     }
+
+    void configureMoveBookmarksDelegate(TestWebExtensionManager *manager)
+    {
+        manager.internalDelegate.moveBookmarkWithIdentifier = ^(NSString *bookmarkId, NSString *toParentId, NSNumber *atIndex, void (^completionHandler)(NSObject<_WKWebExtensionBookmark> *, NSError *)) {
+            NSMutableDictionary *foundInfo = findBookmarkAndParentArrayInMockTree(uiProcessMockBookmarks.get(), bookmarkId);
+            if (!foundInfo) {
+                completionHandler(nil, [NSError errorWithDomain:NSCocoaErrorDomain code:NSExecutableRuntimeMismatchError userInfo:@{ NSLocalizedDescriptionKey: [NSString stringWithFormat:@"Bookmark with ID '%@' not found for move.", bookmarkId] }]);
+                return;
+            }
+
+            NSMutableDictionary *bookmarkToMove = foundInfo[@"bookmark"];
+            NSMutableArray *oldParentChildrenArray = foundInfo[@"parentChildren"];
+
+            NSUInteger oldIndex = [oldParentChildrenArray indexOfObject:bookmarkToMove];
+            if (oldIndex == NSNotFound) {
+                completionHandler(nil, [NSError errorWithDomain:NSCocoaErrorDomain code:NSExecutableRuntimeMismatchError userInfo:@{ NSLocalizedDescriptionKey: @"Bookmark found but not in its reported parent array." }]);
+                return;
+            }
+
+            NSMutableArray *newParentChildrenArray = nil;
+            if (!toParentId)
+                newParentChildrenArray = uiProcessMockBookmarks.get();
+            else {
+                NSMutableDictionary *newParent = findBookmarkInMockTree(uiProcessMockBookmarks.get(), toParentId);
+                if (!newParent || ![newParent[@"type"] isEqualToString:@"folder"] || !newParent[@"children"]) {
+                    completionHandler(nil, [NSError errorWithDomain:NSCocoaErrorDomain code:NSExecutableRuntimeMismatchError userInfo:@{ NSLocalizedDescriptionKey: [NSString stringWithFormat:@"New parent folder with ID '%@' not found or not a folder.", toParentId] }]);
+                    return;
+                }
+                newParentChildrenArray = newParent[@"children"];
+            }
+
+            NSInteger rawTargetIndex = atIndex ? [atIndex integerValue] : newParentChildrenArray.count;
+            [oldParentChildrenArray removeObjectAtIndex:oldIndex];
+
+            if (oldParentChildrenArray == newParentChildrenArray && oldIndex < (NSUInteger)rawTargetIndex)
+                rawTargetIndex--;
+
+            NSInteger finalTargetIndex = rawTargetIndex;
```

Comment:
If this was made an unsigned integer, you wouldn't need the first check on line 454/455.